src/linear_algebra/bilinear_form.lean

```diff
@@ -1697,14 +1705,36 @@ theorem _root_.matrix.nondegenerate.to_bilin {M : matrix ι ι R₃} (h : M.nond
   (b : basis ι R₃ M₃) : (to_bilin b M).nondegenerate :=
 by convert h.to_bilin'.congr b.equiv_fun.symm using 1
 
+lemma _root_.matrix.nondegenerate_iff_to_bilin' {M : matrix ι ι R₃} :
+  M.nondegenerate ↔ (matrix.to_bilin' M).nondegenerate :=
+⟨matrix.nondegenerate.to_bilin', λ h v hv, h v $ λ w, (M.to_bilin'_apply' _ _).trans $ hv w⟩
+
 theorem nondegenerate_of_det_ne_zero' (M : matrix ι ι A) (h : M.det ≠ 0) :
   (to_bilin' M).nondegenerate :=
 (matrix.nondegenerate_of_det_ne_zero h).to_bilin'
 
+lemma to_bilin'_nondegenerate_iff_det_ne_zero {M : matrix ι ι A} :
+  (matrix.to_bilin' M).nondegenerate ↔ M.det ≠ 0 :=
+by rw [←matrix.nondegenerate_iff_to_bilin', matrix.nondegenerate_iff_det_ne_zero]
+
 theorem nondegenerate_of_det_ne_zero (b : basis ι A M₃) (h : (to_matrix b B₃).det ≠ 0) :
   B₃.nondegenerate :=
 to_bilin_to_matrix b B₃ ▸ (matrix.nondegenerate_of_det_ne_zero h).to_bilin b
 
+lemma to_bilin'_nondegenerate_iff_to_bilin_nondegenerate {M : matrix ι ι R₃}
+  (b : basis ι R₃ M₃) : (matrix.to_bilin' M).nondegenerate ↔ (matrix.to_bilin b M).nondegenerate :=
+(congr_nondegenerate_iff b.equiv_fun.symm).symm
+
+lemma nondegenerate_iff_det_ne_zero (B : bilin_form A M₃)
+  (b : basis ι A M₃) : B.nondegenerate ↔ ((bilin_form.to_matrix b) B).det ≠ 0 :=
+begin
+  revert B,
+  rw (bilin_form.to_matrix b).symm.surjective.forall,
+  intro M,
+  rw [to_matrix_symm, to_matrix_to_bilin, ← to_bilin'_nondegenerate_iff_to_bilin_nondegenerate,
+    to_bilin'_nondegenerate_iff_det_ne_zero]
```

Comment:
Even this feels like too much proof - I'd expect we can get there in just two rewrites,
`B.nondegenerate ↔ (bilin_form.to_matrix b B).nondegenerate ↔ (bilin_form.to_matrix b B).det ≠ 0`